internal/server/csp.go

```
@@ -0,0 +1,108 @@
+// the base64-encoded SHA-256 of each inline <script> body it finds.
+//
+// The browser hashes the raw bytes between the opening tag's `>` and the
+// closing `</script>`, including surrounding whitespace — we must hash the
+// exact same substring or the CSP entry will not match.
+func scanInlineScriptHashes(publicDir string) ([]string, error) {
+	if publicDir == "" {
+		return nil, nil
+	}
+	info, err := os.Stat(publicDir)
+	if err != nil || !info.IsDir() {
+		return nil, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 29) but it returns nil

(nilerr)